pkg/app/api/grpcapi/piped_api.go

```diff
@@ -969,11 +970,54 @@ func (a *PipedAPI) UpdateApplicationConfigurations(ctx context.Context, req *pip
 		}
 	}
 	for _, appInfo := range req.Applications {
+		// Retrieve env id by env name.
+		envID := ""
+		for key, value := range appInfo.Labels {
+			if key != config.EnvLabelKey {
+				continue
+			}
+			opts := datastore.ListOptions{
+				Limit: 1,
+				Filters: []datastore.ListFilter{
+					{
+						Field:    "ProjectId",
+						Operator: datastore.OperatorEqual,
+						Value:    projectID,
+					},
+					{
+						Field:    "Name",
+						Operator: datastore.OperatorEqual,
+						Value:    value,
+					},
+					{
+						Field:    "Deleted",
+						Operator: datastore.OperatorEqual,
+						Value:    false,
+					},
+				},
+			}
+			envs, err := a.environmentStore.ListEnvironments(ctx, opts)
+			if err != nil {
+				a.logger.Error("failed to get environments", zap.Error(err))
+				return nil, status.Error(codes.Internal, "Failed to get environments")
+			}
```

Comment:
Just call the list once before entering the loop to fetch all environments of the project to use instead of calling for each application.

Reply:
It should